proto/rill/admin/v1/api.proto

```diff
@@ -2478,6 +2492,17 @@ message ReportOptions {
   repeated string slack_webhooks = 12;
   string web_open_path = 15; // Annotation for the subpath of <UI host>/org/project to open for the report.
   string web_open_state = 14; // Annotation for the base64-encoded UI state to open for the report.
+  string explore = 16; // either canvas or explore should be set
+  string canvas = 17;
+  enum OpenMode {
+    OPEN_MODE_UNSPECIFIED = 0;
+    OPEN_MODE_RECIPIENT = 1; // send this for older reports
+    OPEN_MODE_CREATOR = 2; // send this for reports that should be opened with creators permissions
+    OPEN_MODE_NONE = 3; // send this for reports that should not be opened or older reports which do not have any web_open_path
+    OPEN_MODE_FILTERED = 4; // send this for reports that should be opened with creators permissions but with locked filters
+  }
+  OpenMode web_open_mode = 18;
+  optional rill.runtime.v1.Expression filter = 19; // use with OPEN_MODE_FILTERED
```

Comment:
Again, I don't think `optional` is needed – the empty value will already be `nil`, which I don't think we need to disambiguate further.